test/unit/controllers/CountrySelectModalControllerSpec.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+/* global describe, it, inject, expect, beforeEach, spyOn, jasmine, Config, SearchIndexManager */
+
+describe('CountrySelectModalController', function () {
+  beforeEach(module('myApp.controllers'))
+
+  beforeEach(inject(function (_$controller_, _$rootScope_, ___) {
+    this.$controller = _$controller_
+    this.$rootScope = _$rootScope_
+    this._ = ___
+
+    this.$scope = _$rootScope_.$new()
+    var scope = this.$scope
+    this.createController = function () {
+      this.$controller('CountrySelectModalController', {
+        $scope: scope,
+        $modalInstance: {},
+        $rootScope: _$rootScope_,
+        _: ___
+      })
+    }
+
+    // Spy on IndexObject
+    spyOn(SearchIndexManager, 'indexObject').and.callThrough()
+  }))
+
+  // define tests
+
+  // The tests before controller initiation.
+  // In order to mock Config data
+
+  it('initiates Country to select', function (done) {
+    Config.CountryCodes = [['NL', 'country_select_modal_country_nl', '+31']]
+    var expected = 'NL Netherlands +31'
+
+    this.createController()
+
+    expect(SearchIndexManager.indexObject).toHaveBeenCalledWith(0, expected, jasmine.any(Object))
+    done()
+  })
+
+  it('initiates Countriy to select with 2 (or more) country codes', function (done) {
+    Config.CountryCodes = [['VA', 'country_select_modal_country_va', '+39 06 698', '+379']]
+    var expected = 'VA Vatican City +39 06 698 +379'
+
+    this.createController()
+
+    expect(SearchIndexManager.indexObject).toHaveBeenCalledWith(0, expected, jasmine.any(Object))
+    done()
+  })
+
+  it('initiates Countries to select', function (done) {
+    Config.CountryCodes = [['NL', 'country_select_modal_country_nl', '+31'], ['VA', 'country_select_modal_country_va', '+39 06 698', '+379']]
+    var expected1 = 'NL Netherlands +31'
+    var expected2 = 'VA Vatican City +39 06 698 +379'
```

Comment:
Since you start to repeat them here, I'd suggest to move them to a `beforeAll` block and make them „magic constants”. Like `this.VATICAN_COUNTRY_CODE`.